src/UglyToad.PdfPig/Geometry/PdfPath.cs

```diff
@@ -18,6 +18,11 @@ public class PdfPath
         /// </summary>
         public IReadOnlyList<IPathCommand> Commands => commands;
 
+        /// <summary>
+        /// True if the <see cref="PdfPath"/> was originaly draw as a rectangle.
+        /// </summary>
+        internal bool IsDrawnAsRectangle { get; set; }
```

Comment:
I'm fine with this appearing on the public API since it's useful information. A better name might be `IsPartOfRectangle` maybe? Not really sure. (also remove the public setter please :) )

Reply:
Agree on making it `public` but I would keep the name as only a rectangle (and not a part) can have this flag set a `true`. (Especially true for clipping rectangles)